### ISSUE TITLE
.js is a 2 character TLD and those are resevered for countries

### ISSUE THREAD

**Issue by ozten** [ozten/TLD.js#8]
as reported in. https://twitter.com/#!/bdufresne/status/204643709441159168

Is this accurate? Any way around it?

http://www.newgtldsite.com/dot-code-tld-domain-names/

**Comment by jesstelford** [comment]
+1 for `.src` :D

**Comment by redaktor** [comment]
What exactly defines a ```country``` ?
For Instance how about unincorporated unorganized territories:
American Samoa has got its own TLD …
How about .js for Jarvis Island ?

**Comment by ozten** [comment]
Is Liberland a new opportunity?
http://en.wikipedia.org/wiki/Liberland

Anyone have an connections with Vít Jedlička or other leaders of the country?

**Comment by timbur** [comment]
.jsx imo!

**Comment by indus** [comment]
JS.ORG provides free subdomains to The JavaScript community (see [dns.js.org](http://dns.js.org))

**Comment by mcanthony** [comment]
The idea proposed by @hyrmedia  is not all that unreasonable, it may be possible to propose the local government of a small nation, one which has not yet a TLD, to sponsor a .JS app - if it would not cost them and it would in fact generate funds for said nation. This would of course put that nations government in control of the TLD at the end of the day, and not the JS/Dev community, though.

If we where to get .SRC, would we not be able to add SLDs (js.src) to at will such as @Zederich described? Once everyone settled on a TLD, that is not reserved/unavailable, I think it would be wise to  create an open governance run repo specifically for this purpose, so as to broaden the potential interested parties, and avoid the inevitable hate that would come from NON-JS developers *cough* @JarrettBillingsley *cough*.

A while back I did some research on ICANN regs and I recall something about the applicant/entity must have funds/holdings (not including assets) of or exceeding something around $1M (I cant remember). If I recall this requirement was to ensure the applicant has sufficient resources to see it through.

On the subject of gTLDs, $2M has been set aside by ICANN to assist "needy" applicants. http://newgtlds.icann.org/en/applicants/candidate-support/faqs - this might be a ray of light in a long, dark and expensive hallway. If it becomes largely unreasonable to think that a gTLD is attainable, perhaps an existing gTLD has a suitable name that happens to co-incide with our tastes - and if this gTLD owner would be willing to sell SLD rights to us at a greatly reduced cost and without having to suffer the wrath of ICANN. Since there are thousands of new gTLD owners, I am sure many are suffering and willing to strike all kinds of weird deals. :)

I propose that this project center around the idea of becoming a non-profit OSS organization which intends to use community support and donations, and domain sales to sustain it's operation. The operations primary goal should be to seek to generate funds in which proceeds would go toward education related charities - perhaps one that centers around education in the web development sector. This would be a good model to gain support internationally, as many developing countries (Vietnam a good example) will surely emerge as a source of new talent in this growing industry. And changes in the direction of the way the american government deals with trading internationally coincide with this notion.

This could be a small piece in a big puzzle that makes up the new global economy - driven by the connected web.

**Comment by mikeal** [comment]
> I propose that this project center around the idea of becoming a non-profit OSS organization which intends to use community support and donations, and domain sales to sustain it's operation.

We've a few OSS non-profits already. I'm tied in to the Node Foundation and know quite a few people at the jQuery Foundation and the Mozilla Foundation. Additionally, the Linux Foundation spins up new non-profits and offers a lot of services to keep the costs down (this week they spun up the Node Foundation and Open Container Project where Docker now resides). If the blocker is having a non-profit behind this we have more than enough.

**Comment by redaktor** [comment]
just pinging 
The Dojo Foundation @dojo @dylans due to the comment before (and this fascinating debate).

**Comment by mcanthony** [comment]
@mikeal I don't think it's a blocker but it might be a door opener, one that would surely draw in more support/attention than simply a small group of people wanting for a cool domain. So perhaps the next step could be to gain the support of one of these plentiful OSS NPOs.

**Comment by horaceho** [comment]
How about `.git`?

    git clone angular.js.git
    git clone django.py.git
    git clone boost.cpp.git
    ...

looks like a nice place to keep codes ;-)

The domain names can be reserved for top starred GitHub repos to avoid zoombie domains.

**Comment by thenoncoder** [comment]
`.py' domain name already exists, but, IIRC, only people currently living in Paraguay can register for one.
Oh and btw, does anyone want to help me make a country/island called `Javascriptia`?
I can provide you with a construction company, and you provide me with the money required. I'm actually serious about this.

**Comment by michaelzoidl** [comment]
In the first step we could buy the `src.io` domain... 
i think `ruby.src.io/rails` doesnt sound that bad... or `js.src.io/react`

**Comment by adriancooney** [comment]
The folks over at https://js.org/ are doing some good work in line with this thread. You can't get domains like http://jquery.js.org.

**Comment by DaKnOb** [comment]
Guys, guys.. What if you think outside the box? What if you... Think Different™?

It's probably easier to change the `.js` extension globally (browsers, web servers, projects, etc.) than to "start" a new country. 

Aside from that, trying to convince ICANN to make an exception / find a 3+ letter domain is probably an option as well.

**Comment by Boldewyn** [comment]
@DaKnOb so you basically suggest a dark web for web developers? Hmm, sounds interesting, I'm curious how that would work out... :smile:

**Comment by DaKnOb** [comment]
@Boldewyn Well, what I suggest is `%/\.js/\.javascript/g` (or something). Rename *every* JavaScript file to another extension, then use the money to get *that* gTLD and convince browsers / other devs / projects to use the new extension for JavaScript files, instead of the `.js` one.

**Comment by Boldewyn** [comment]
Ah, ok. Got it.

**Comment by arxpoetica** [comment]
Dark magic?

![image](https://cloud.githubusercontent.com/assets/111424/12267331/9a99058e-b915-11e5-80e9-51e86c54c2b8.png)

**Comment by michaelzoidl** [comment]
It's easy... let's develop a browser which handles the `.js` TLD... 
![](https://s-media-cache-ak0.pinimg.com/originals/b2/8a/3a/b28a3a52ee99399a5389e758f7de87b6.gif)

**Comment by dodekeract** [comment]
@michaelzoidl I think you meant let's make our own DNS server, right?

**Comment by ozten** [comment]
Sure, [_why](https://en.wikipedia.org/wiki/Why_the_lucky_stiff) did this a decade ago. Go for it!

But it isn't a solution to this blocker. Anyone, without installing any special software or configurations should be able to resolve a .js domain.

**Comment by thenoncoder** [comment]
Why isn't this closed already? It won't happen.

**Comment by lassediercks** [comment]
bump

**Comment by JemiloII** [comment]
TL;DR; We should pair with google, opennicproject.org, cloudflare, and other dns to make our own tld. This way we can get our lovely .api, .js, or .etc We don't need icann.

I know this post has been out there for a while, but as developers, we really should join up with groups like opennicproject.org and create our own tld. ICANN is corrupt and they allow for new tlds to charge with no cap. If it wasn't for the US government stepping in, verisign would have the .com registration to be around $10. This would just be the verisign fee.

Have you ever wondered about the cost of a domain name?
ICANN: $0.18 (Flat)
Verisign -- the .com registry: $7.85 (Capped by US gov)
3% credit card processing fee: $0.30+
Your registrar: $1.67+

So why does verisign want to increase prices? So they can keep the same net revenue. What's funny? Their maintenance cost are low as they require their partner registrars to host copies of their registrar on their systems.

So you pour out your hard work on a cake api. You want cake.api, but because of ICANN's flimsy rules and high requirements to manage and maintain a tld, it's the stupid rules requiring expensive insurance and $70k in liquid cash, with $4k application fee. <--- Say what?!

You end up realizing that cake.api cost  $1000 USD because it has no price caps, because the new registrar wants to make their investment back, because its a 4 letter domain.

So you put a donation button on your github, and your api is popular, and you raise the money to learn that some Chinese investor purchased the domain and is doing nothing with it except to resale it back to you for $100k, it sounds ridicules, but this actually does happen. Just google about the investors who believe they have a billion dollar net worth domain name portfolio. Or talk to godaddy support and wait for the support team to offer a $300k domain to you to see if you bite.

It's a sick reality and really we should be proactively do something about this crisis. Lets say, you really wanted cake.api, raised 100k+ to hire lawyers, translators, and to purchase the domain, to find out that cakes.com purchased cake.api so that it would just point to cakes.com. At that point, you're donzo washington and have to go back to your firepit to listen to the voice of Morgan Freeman to calm yourself down and give you new encouragement with your 100k raised.

But hey,  if you want to take the easy path out and change your api name to someuntastyfood.api, then be a pushover.

So what if icann wants to sell our tld? Well, lets register several business. Get one in the US, and some in other countries. We can just sue them when they attempt to take our property. Icann does not own the web. We can users do. Worried about funds? Lets just let users know upfront that we have a legal fee to protect their domain. We can have our own light weight rules such as: No up/reselling domains, Domain must be in use, and in the case of .api, domain must have an api. These aren't fixed rules, these are just examples.

A lot of this post may sound like a rant, but just following the money, there is a huge monopoly that is not in our interest. I remember when a tld was suppose to have a purpose, .com was to be commerical, however, its more than just that. Its really an .anything domain.

**Comment by WebReflection** [comment]
so, to sum it up:

> I know this post has been out there for a while ...
> A lot of this post may sound like a rant ...

I guess this TLD ain't gonna happen so that *maybe*, unless it does or there is some concrete news about it, people could stop necrobumping this issue?

Thank You!

**Comment by dodekeract** [comment]
@WebReflection Just unsubscribe if you don't want to receive notifications.

**Comment by DaKnOb** [comment]
@JemiloII 
It is not as simple as that. First of all, what makes you think Google will side with you on this just so you can have your own .js domain name about a project you made? Also, even if they do, will all the ISPs who operate name servers? 
Second of all, operating the .com root zone does not come cheap. It is not a $5 VM on Digital Ocean that runs bind. Do you know how many servers there are, in how many locations, and how many people working 24/7 behind this? Do you know how much this costs? 
Finally, what makes you think ICANN is bad and that we should bypass them because domains belong to the users? Can I, in a similar manner, advertise any IP Range I want, bypassing IANA, because IP Addresses are a public resource so they belong to me, too? These organizations have been created to avoid chaos and they may not be perfect, but by limiting users a little bit, provide so much more.

**Comment by WebReflection** [comment]
@dodekeract I know it sounded rude and I wrote in a rush but try to imagine I subscribed because I'm interested like anyone else in here but the amount of notifications about GIFs, bumps, rants, and everything else that won't help reached kinda the highest peak, IMO ... so no, I want to be notified and I hope daily that next time it will be for some concrete news about this proposal.

Best Regards

**Comment by thenoncoder** [comment]
.js TLD is probably never going to happen, unless in very exceptional cases it does.

I suggest to close this issue, because the only other thing is bypassing ICANN, and obviously that should never be done.

**Comment by MuneCms** [comment]
.js can't happen  .porn and .gay are available but .africa is not!

**Comment by JohnnyCache42** [comment]
+1 for .src and/or .script.

What about .gnu, .copyleft, or .ccl (creative commons license)?

btw - I couldn't find anything that says that a 2 letter second level domain was not allowed, so .js.src should work....

**Comment by indus** [comment]
@JohnnyCache42 Thats what [JS.ORG](http://js.org) does.

**Comment by thenoncoder** [comment]
Close this issue, please.

**Comment by daniellmb** [comment]
http://js.co.de/ is still available for 12 Euro :)

**Comment by dodekeract** [comment]
@daniellmb looks like a scam site though [source](https://www.mywot.com/en/scorecard/js.co.de?utm_source=addon&utm_content=rw-viewsc)

**Comment by cybernet** [comment]
HA !

Get Outlook for iOS




On Fri, Jun 10, 2016 at 3:41 AM +0300, "Daniel Lamb" <notifications@github.com> wrote:












http://js.co.de/ is still available for 12 Euro :)

**Comment by l0oky** [comment]
Sounds like a great investment to create a country with JS initials. 
That country would get rich fast :D

**Comment by aamironline** [comment]
@ozten I liked the idea of `.dev` - Developer friendly and generic enough to satisfy everyone's needs!

**Comment by dodekeract** [comment]
@aamironline Also works as a dev(ice) domain for internet of things stuff.

**Comment by mark0978** [comment]
What about .es6?  Gets around the 2 character problem, unfortunately .es7 makes it obsolete unless we pull the html5 stunt and just decide we don't need versions after 5

**Comment by drsounds** [comment]
Why cant we create a virtual countrt for all devs which has the code js?

**Comment by christian-bromann** [comment]
+1 👍  any updates on this?

**Comment by mark0978** [comment]
The PitCairn Islands are currently .pn and could use the money
If someone could convince them to change their name to something that would give us .js.....
Maybe .js could be Sea of Japan similar to .io being Indian Ocean.

**Comment by drsounds** [comment]
I don't think we should risk a world war 3 with nuclear holocaust because
of the developer community wishes. The world politics is too unstable at
this moment to test the boundaries....

Best regards,
Alexander Forselius (Dr. Sounds)
Musician, and programmer.
http://dr-sounds.com

Github
<http://t.sidekickopen06.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1q0Mw-W8r4PxK7fsJrMW3N1zH656dR01f5JhpKW02?t=http%3A%2F%2Fwww.github.com%2Fkrikelin&si=5411275823644672&pi=52668902-d4d3-47a8-e561-b845113fae7e>
Twitter
<http://t.sidekickopen06.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1q0Mw-W8r4PxK7fsJrMW3N1zH656dR01f5JhpKW02?t=http%3A%2F%2Fwww.twitter.com%2Fdrsounds&si=5411275823644672&pi=52668902-d4d3-47a8-e561-b845113fae7e>
Spotify
<http://t.sidekickopen06.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1q0Mw-W8r4PxK7fsJrMW3N1zH656dR01f5JhpKW02?t=http%3A%2F%2Fopen.spotify.com%2Fartist%2F2FOROU2Fdxew72QmueWSUy&si=5411275823644672&pi=52668902-d4d3-47a8-e561-b845113fae7e>
<http://t.sidekickopen06.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1q0Mw-W8r4PxK7fsJrMW3N1zH656dR01f5JhpKW02?t=http%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3Dpjdujamrcds&si=5411275823644672&pi=52668902-d4d3-47a8-e561-b845113fae7e>
LinkedIn
<http://t.sidekickopen06.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs1q0Mw-W8r4PxK7fsJrMW3N1zH656dR01f5JhpKW02?t=http%3A%2F%2Fse.linkedin.com%2Fpub%2Falexander-forselius%2F13%2Fb11%2F500%2F&si=5411275823644672&pi=52668902-d4d3-47a8-e561-b845113fae7e>


2016-12-07 17:07 GMT+01:00 Mark Jones <notifications@github.com>:

**Comment by DaKnOb** [comment]
@christian-bromann Good news! The U.S. State of California decided to split from the United States and is willing to rename itself to "California.js". They will be applying for the `.js` TLD as soon as this is announced and official.

**Comment by ozten** [comment]
**Please do not comment on this** issue, except in the very unlikely case that you have something to unblock this issue. Comments that are off topic, ideas that have already been considered, etc will be deleted.

**Comment by matyapiro31** [comment]
Use OpenNIC and New Nations.
https://www.opennic.org/
http://new-nations.net/en/

**Comment by rsp** [comment]
Some other options to consider:

- General JavaScript: `.undefinedisnotafunction`
- Node-specific:` .eventedioforv8javascript` (anyone still remembers?)
- Frontend-specific: `.windowdocumentgetelementsbyclassname` or `.$`

**Comment by Arshiamidos** [comment]
.null
.undefined
.fuck

**Comment by petersaints** [comment]
if it has to have 3 letters, what about ".jsc"?
